webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js

```diff
@@ -658,6 +654,15 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
 
     if (currentIndex >= 1) {
       this._hydratePrevious(currentIndex);
+
+      // Here we should save the current data of the submission and change to the next one
```

Comment:
What I would do is add this logic to the submission setter. If you look at set _submission you will see I did lots of state changes in that setter. We should keep the logic in one place. If a bug is related to changing the submission, change the code that sets the submission.